src/include/optimizer/optimize_context.h

```diff
@@ -12,27 +12,37 @@
 
 #pragma once
 
+#include "common/timer.h"
+
 #include "optimizer/property_set.h"
 #include "optimizer/optimizer_task.h"
 #include "optimizer/optimizer_task_pool.h"
 
+#include "settings/settings_manager.h"
+
 namespace peloton {
 namespace optimizer {
 
 class OptimizerMetadata;
 
 class OptimizeContext {
  public:
-  OptimizeContext(OptimizerMetadata *metadata,
-                  std::shared_ptr<PropertySet> required_prop,
-                  double cost_upper_bound = std::numeric_limits<double>::max())
+  OptimizeContext(
+      OptimizerMetadata *metadata, std::shared_ptr<PropertySet> required_prop,
+      double cost_upper_bound = std::numeric_limits<double>::max(),
+      unsigned int timeout_limit = settings::SettingsManager::GetInt(
+          settings::SettingId::task_execution_timeout))
       : metadata(metadata),
         required_prop(required_prop),
-        cost_upper_bound(cost_upper_bound) {}
+        cost_upper_bound(cost_upper_bound),
+        timeout_limit(timeout_limit),
+        timer(Timer<std::milli>()) {}
 
   OptimizerMetadata *metadata;
   std::shared_ptr<PropertySet> required_prop;
   double cost_upper_bound;
+  unsigned int timeout_limit;
```

Comment:
Could we put members related to timeout into `OptimizerMetadata`? `OptimizerMetadata` is kept per query, while `OptimizeContext` is kept for each operator.

Reply:
done